RecoMET/METPUSubtraction/plugins/NoPileUpPFMEtProducer.cc

```diff
@@ -360,27 +347,26 @@ void NoPileUpPFMEtProducer::produce(edm::Event& evt, const edm::EventSetup& es)
   initializeCommonMETData(*sumPUjets);
   std::vector<metsig::SigInputObj> metSignObjectsPUjets;
   int jetIdx = 0;
-  for ( reco::MVAMEtJetInfoCollection::const_iterator jet = jets_cleaned.begin();
+  for ( reco::PUSubMETCandInfoCollection::const_iterator jet = jets_cleaned.begin();
 	jet != jets_cleaned.end(); ++jet ) {
-    //if ( verbosity_ ) printMVAMEtJetInfo("jet", jetIdx, *jet);
-    if ( jet->passesLooseJetId_ ) {
-      if ( jet->type_ == reco::MVAMEtJetInfo::kNoPileUp ) {	
-	addToCommonMETData(*sumNoPUjets, jet->p4_);
-	metSignObjectsNoPUjets.push_back(jet->pfMEtSignObj_);
-	sumNoPUjetOffsetEnCorr->mex   += jet->offsetEnCorr_*cos(jet->p4_.phi())*sin(jet->p4_.theta());
-	sumNoPUjetOffsetEnCorr->mey   += jet->offsetEnCorr_*sin(jet->p4_.phi())*sin(jet->p4_.theta());
-	sumNoPUjetOffsetEnCorr->mez   += jet->offsetEnCorr_*cos(jet->p4_.theta());
-	sumNoPUjetOffsetEnCorr->sumet += jet->offsetEnCorr_*sin(jet->p4_.theta());
+    if ( jet->passesLooseJetId() ) {
+      if ( jet->type() == reco::PUSubMETCandInfo::kHS ) {	
+	addToCommonMETData(*sumNoPUjets, jet->p4() );
+	metSignObjectsNoPUjets.push_back(jet->metSignObj() );
+	sumNoPUjetOffsetEnCorr->mex   += jet->offsetEnCorr()*cos(jet->p4().phi())*sin(jet->p4().theta());
```

Comment:
better not use the practice of computing px/p by going through arccos, cos, arctan etc, without complicated functions:
``` C++
float jetp = jet->p4().p();
float jetcorr = jet->offsetEnCorr();
sumNoPUjetOffsetEnCorr->mex +=  jetcorr* jet->p4().px()/jetp;

```